pages/index.js

```diff
@@ -217,27 +234,38 @@ export default function HomePage({
         </StyledControls>
       </StyledSelectionBar>
 
+      <StyledTimelineFilterContainer>
+        <TimelineFilter
+          selectedTimeframe={selectedTimeframe}
+          onTimeframeChange={handleTimeFrameChange}
+        />
+      </StyledTimelineFilterContainer>
+
       <StyledSortContainer>
         <SortControl
           sortOrder={sortOrder}
           onToggleSortOrder={handleToggleSortOrder}
         />
       </StyledSortContainer>
 
-      <TransactionsList
-        handleEditTransaction={handleEditTransaction}
-        transactions={displayedTransactions}
-        selectedCategory={selectedCategory}
-        handleOpenEditMode={handleOpenEditMode}
-        openModal={openModal}
-        onCloseModal={closeModal}
-        handleOpenDeleteDialogue={handleOpenDeleteDialogue}
-        handleConfirmDelete={handleConfirmDelete}
-        handleCancelDeleteDialogue={handleCancelDeleteDialogue}
-        handleDeleteTransaction={handleDeleteTransaction}
-        isDeletingId={isDeletingId}
-        sortOrder={sortOrder}
-      />
+      {displayedTransactions.length > 0 ? (
+        <TransactionsList
+          handleEditTransaction={handleEditTransaction}
+          transactions={displayedTransactions}
+          selectedCategory={selectedCategory}
+          handleOpenEditMode={handleOpenEditMode}
+          openModal={openModal}
+          onCloseModal={closeModal}
+          handleOpenDeleteDialogue={handleOpenDeleteDialogue}
+          handleConfirmDelete={handleConfirmDelete}
+          handleCancelDeleteDialogue={handleCancelDeleteDialogue}
+          handleDeleteTransaction={handleDeleteTransaction}
+          isDeletingId={isDeletingId}
+          sortOrder={sortOrder}
+        />
+      ) : (
+        <p>No transactions found within this timeframe.</p>
```

Comment:
there are other possiblities for the transaction list being empty. Please adapt the text such that it more general like "No transactions found".